src/duct.py

```diff
@@ -164,62 +166,165 @@ def create_and_parse_args():
         default=60.0,
         help="Interval in seconds at which to report aggregated data.",
     )
+    parser.add_argument(
+        "--capture-outputs",
+        type=str,
+        default="all",
+        choices=["all", "none", "stdout", "stderr"],
+        help="Record stdout, stderr, all, or none to log files.",
+    )
+    parser.add_argument(
+        "--outputs",
+        type=str,
+        default="all",
+        choices=["all", "none", "stdout", "stderr"],
+        help="Print stdout, stderr, all, or none to stdout/stderr respectively.",
+    )
+    parser.add_argument(
+        "--record-types",
+        type=str,
+        default="all",
+        choices=["all", "system-summary", "processes-samples"],
+        help="Record system-summary, processes-samples, or all",
+    )
     return parser.parse_args()
 
 
+class TeeStream:
+    """TeeStream simultaneously streams to standard output (stdout) and a specified file."""
+
+    def __init__(self, file_path):
+        self.file = open(file_path, "w")
+        (
+            self.listener_fd,
+            self.writer_fd,
+        ) = os.openpty()  # Use pseudo-terminal to simulate terminal behavior
+
+    def fileno(self):
+        """Return the file descriptor to be used by subprocess as stdout/stderr."""
+        return self.listener_fd
+
+    def start(self):
+        """Start a thread to read from the main_fd and write to stdout and the file."""
+        thread = threading.Thread(target=self._redirect_output, daemon=True)
+        thread.start()
+
+    def _redirect_output(self):
+        with os.fdopen(self.listener_fd, "rb", buffering=0) as stream:
```

Comment:
is there any performance impact if there is heavy IO?  e.g. run with some dummy `cat largefile` and see CPU ... is it less of effect if buffering is enabled?